libs/ui/elements/src/lib/record-api-form/record-api-form.component.html

```diff
@@ -37,15 +37,37 @@
           </div>
         </div>
       </div>
-      <div class="flex flex-col gap-3">
-        <p class="text-sm" translate>record.metadata.api.form.offset</p>
-        <gn-ui-text-input
-          class="w-20"
-          [value]="offset$ | async"
-          (valueChange)="setOffset($event)"
-          hint=""
-        >
-        </gn-ui-text-input>
+      <div class="flex flex-col gap-3 relative">
+        <p class="text-sm" [class.text-gray-600]="!supportOffset" translate>
+          record.metadata.api.form.offset
+        </p>
+        <div class="flex items-center">
+          <gn-ui-text-input
+            class="w-20"
+            [value]="offset$ | async"
+            [extraClass]="
+              supportOffset ? '' : 'text-gray-600 bg-gray-200 opacity-70'
+            "
```

Comment:
can't you simply make this component disabled instead of giving it css classes?

Reply:
I don't think that's possible as the `gn-ui-text-input` component does not have a property to disable.